scripts/routing.js

```diff
@@ -128,34 +135,42 @@ async function calculateFullRoute(fromCoordinates, toCoordinates) {
 	let coords1 = ol.proj.fromLonLat([routeSummaryBike.bbox[0], routeSummaryBike.bbox[1]]);
 	let coords2 = ol.proj.fromLonLat([routeSummaryBike.bbox[2], routeSummaryBike.bbox[3]]);
 	let convertedBbox = [coords1[0], coords1[1], coords2[0], coords2[1]];
-	let viewableBox = [map.getSize()[0], map.getSize()[1] - document.getElementById("placeSearchMenu").clientHeight];
+	let menuHeight;
+	if (document.getElementById("placeSearchMenu")) menuHeight = document.getElementById("placeSearchMenu").clientHeight;
+	else if (document.getElementById("stationMenu"))
+		menuHeight = document.getElementById("stationMenu").clientHeight + 30;
+	else menuHeight = 0;
```

Comment:
Acho que se pode simplificar isto um pouco mas se achares que fica menos legível podes deixar. Se aceitares não te esqueças de correr prettier depois
```suggestion
	const menuHeight = document.getElementById("placeSearchMenu")?.clientHeight ??(document.getElementById("stationMenu")?.clientHeight + 30) || 0
```

Reply:
acho que o meu small brain não consegue compreender isto a 100%